packages/core/src/analytics/analytics-extra-items-manager.ts

```diff
@@ -0,0 +1,54 @@
+import { LineItemMap } from '../cart';
+import { ExtraItemsData } from './analytics-step-tracker';
+
+const ORDER_ITEMS_STORAGE_KEY = 'ORDER_ITEMS';
+
+export default class AnalyticsExtraItemsManager {
+    constructor(
+        private storage: StorageFallback,
+    ) {}
+
+    saveExtraItemsData(id: string, lineItems: LineItemMap): ExtraItemsData {
+        const data = [
+            ...lineItems.physicalItems,
+            ...lineItems.digitalItems,
+        ].reduce((result, item) => {
+            result[item.productId] = {
+                brand: item.brand ? item.brand : '',
+                category: item.categoryNames ? item.categoryNames.join(', ') : '',
+            };
+
+            return result;
+        }, {} as ExtraItemsData);
```

Comment:
Can we not use type casting here? And use typeguards if necessary?

Reply:
@animesh1987 
We can use another syntax for this
Something like that `reduce<ExtraItemsData>`

I think typeguards are unnecessary here

Reply:
@animesh1987 changed to `reduce<ExtraItemsData>`